src/content/tutorials/build-a-rust-markdown-parser-webassembly/index.md

```diff
@@ -0,0 +1,156 @@
+---
+updated: 2020-09-12
+difficulty: Advanced
+---
+
+# Build a Markdown Parser with Rust and WebAssembly
+
+This tutorial will walk you through the steps of generating, building, previewing, configuring, and publishing
+a Rust-generated WebAssembly serverless function that parses Markdown for Cloudflare Workers. Let's get started!
+
+This tutorial makes use of [Wrangler](https://github.com/cloudflare/wrangler), our command-line tool for generating, building, and publishing projects on the Cloudflare Workers platform. If you haven't used Wrangler, we recommend [installing Wrangler](/cli-wrangler/install-update), or checking out our [Getting Started guide](/learning/getting-started), which will get you set up with Wrangler, and familiar with the basic commands.
+
+## Generating from template
+
+Cloudflare's command-line tool for managing Workers projects, Wrangler, has great support for templates – pre-built collections of code that make it easy to get started writing Workers. We'll make use of the [rustwasm-worker template](https://github.com/cloudflare/rustwasm-worker-template/) to start building your project.
+
+In the command line, generate your Workers project by passing in a project name (we'll use rustwasm-markdown-parser), and the template URL to base your project on.
+
+```sh
+wrangler generate rustwasm-markdown-parser https://github.com/cloudflare/rustwasm-worker-template/
+```
+
+This creates a directory called `rustwasm-markdown-parser` which you can now `cd` into.
+
+Wrangler templates are just Git repositories, so if you want to create your own templates, or use one from our [Template Gallery](/templates), there's a ton of options to help you get started.
+
+## Testing locally
+
+You can test how your Workers function will look when it's deployed by using the preview service, which you can access with the `wrangler dev` command:
+
+```sh
+wrangler dev
+```
+
+[`wrangler dev`](/cli-wrangler/commands#dev) starts a server on localhost that executes your Worker on incoming HTTP requests. By default, it'll listen on port :8787. 
+
+<Aside>The first time you run `wrangler dev` might take a couple minutes to compile to WebAssembly. </Aside>
+
+When navigating to localhost:8787 in your browser, you should see something like this:
+
+![Hello wasm-worker](./media/hello-wasm-worker.png)
+
+
+## Building your code
+
+Let's make our Workers function more interesting. We'll pull in a dependency from the `crates.io` ecosystem called `pulldown-cmark`.
+We'll add this to our `Cargo.toml`:
+
+```toml
+## Cargo.toml
+
+[dependencies]
+pulldown-cmark = "0.4.0"
+```
+
+Now we'll leverage the code in the `string-to-string` example from the `pulldown-cmark` GitHub repository. Let's change
+our `src/lib.rs` to look like this:
+
+```rust
+// src/lib.rs
```

Comment:
Use filenames feature of code blocks (see https://docs-engine-docs.adam.workers.dev/markdown#filenames).